### PR TITLE
Add backpressure support which retains last Query value.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
_Might_ fix #34, but can't repro it.

File: sqlbrite/src/main/java/com/squareup/sqlbrite/BackpressureBufferLastOperator.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          return;
+        }
+
+        Object candidate;
+        synchronized (BufferLastSubscriber.this) {
+          candidate = last;
+
+          long currentRequested = requested;
+          if (Long.MAX_VALUE - n <= currentRequested) {
+            requested = Long.MAX_VALUE;
```
Comment (mattprecious) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BackpressureBufferLastOperator.java]: So... if `requested` is going to overflow, open the firehose?

Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BackpressureBufferLastOperator.java]: Yes. Long.MAX_VALUE is basically "IDGAF just onNext me until you blow up"

File: sqlbrite/src/main/java/com/squareup/sqlbrite/BackpressureBufferLastOperator.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        }
+      }
+
+      if (emit) {
+        child.onNext(t);
+      }
+    }
+
+    @Override public void onStart() {
+      request(Long.MAX_VALUE);
```
Comment (mattprecious) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BackpressureBufferLastOperator.java]: Doesn't this just perma-kill the backpressure logic?

```java
long currentRequested = requested;
if (Long.MAX_VALUE - n <= currentRequested) {
  requested = Long.MAX_VALUE;
}
```

This sets `requested` to `Long.MAX_VALUE` and then this statement always evaluates to true going forward. No?

Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BackpressureBufferLastOperator.java]: This isn't for us. It's for upstream.

**Comment by mattprecious** [comment]
LGTM

**Comment by JakeWharton** [comment]
I'd like to play with changing this to do compare-and-swaps instead of locks, but for now the behavior is all I want.

**Pull request merged**